control/segreg/ignore.go

```diff
@@ -0,0 +1,177 @@
+// Copyright 2025 Anapaya Systems
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package segreg
+
+import (
+	"context"
+	"strings"
+	"text/template"
+
+	"github.com/go-viper/mapstructure/v2"
+
+	"github.com/scionproto/scion/control/beacon"
+	"github.com/scionproto/scion/pkg/addr"
+	"github.com/scionproto/scion/pkg/log"
+	"github.com/scionproto/scion/pkg/private/serrors"
+)
+
+// logLevel defines the logging level for the IgnoreSegmentRegistrationPlugin.
+type logLevel string
+
+const (
+	LogLevelDebug logLevel = "debug"
+	LogLevelInfo  logLevel = "info"
+	LogLevelError logLevel = "error"
+)
+
+func parseLogLevel(s string) (logLevel, error) {
+	switch s {
+	case string(LogLevelDebug), string(LogLevelInfo), string(LogLevelError):
+		return logLevel(s), nil
+	default:
+		return "", serrors.New("invalid log level", "level", s)
+	}
+}
+
+// pluginConfig holds the configuration for the IgnoreSegmentRegistrationPlugin.
+type pluginConfig struct {
+	Level   logLevel
+	Message *template.Template
+}
+
+func parseConfig(config map[string]any) (pluginConfig, error) {
+	// Decode the map.
+	var parsedConfig struct {
+		Level   string
+		Message string
+	}
+	err := mapstructure.Decode(config, &parsedConfig)
+	if err != nil {
+		return pluginConfig{}, serrors.Wrap("decoding plugin configuration", err,
+			"config", config)
+	}
+	// Default log level is debug.
+	if parsedConfig.Level == "" {
+		parsedConfig.Level = string(LogLevelDebug)
+	}
+	// Extract the log level and message from the config.
+	level, err := parseLogLevel(parsedConfig.Level)
+	if err != nil {
+		return pluginConfig{}, serrors.Wrap("parsing log level", err,
+			"value", parsedConfig.Level)
+	}
+	var message *template.Template
+	if parsedConfig.Message != "" {
+		message, err = template.New("message").Parse(parsedConfig.Message)
+		if err != nil {
+			return pluginConfig{}, serrors.Wrap("parsing message template", err,
+				"value", parsedConfig.Message)
+		}
+	}
+	return pluginConfig{Level: level, Message: message}, nil
+}
+
+type IgnoreSegmentRegistrationPlugin struct{}
+
+var _ SegmentRegistrationPlugin = (*IgnoreSegmentRegistrationPlugin)(nil)
+
+func (p *IgnoreSegmentRegistrationPlugin) ID() string {
```

Comment:
```suggestion
func (IgnoreSegmentRegistrationPlugin) ID() string {
```

these methods don't need to be a pointer receiver since the type is `struct{}`.